agentops/session.py

```diff
@@ -43,21 +46,52 @@ def set_session_video(self, video: str) -> None:
         """
         self.video = video
 
+    def add_metadata(self, key: str, value: Any) -> None:
+        """
+        Adds metadata to the session.
+
+        Args:
+            key (str): The metadata key.
+            value (Any): The metadata value.
+        """
+        if self.metadata is None:
+            self.metadata = {}
+        self.metadata[key] = value
+
     def end_session(
         self, end_state: str = "Indeterminate", end_state_reason: Optional[str] = None
     ) -> None:
         """
-        End the session with a specified state, rating, and reason.
+        End the session with a specified state and reason.
 
         Args:
             end_state (str, optional): The final state of the session. Options: "Success", "Fail", "Indeterminate"
-            rating (str, optional): The rating for the session.
             end_state_reason (str, optional): The reason for ending the session. Provides context for why the session ended.
         """
         self.end_state = end_state
         self.end_state_reason = end_state_reason
         self.end_timestamp = get_ISO_time()
 
+    def expire_session(self) -> None:
+        """
+        Mark the session as expired.
+        """
+        self.is_expired = True
```

Comment:
we probably dont need a separate flag for expiring. we currently check `end_state` to see if it is not None, and if not, then it must have ended. so `end_state` could just be `expired` to denote that it is expired.